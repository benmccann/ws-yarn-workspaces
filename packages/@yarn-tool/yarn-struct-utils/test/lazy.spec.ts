//@noUnusedParameters:false

import { basename, extname } from 'path';
import { parseResolutionOrDescriptor } from '../index';

describe(basename(__filename, extname(__filename)), () =>
{
	[
		`@strictsoftware/typedoc-plugin-monorepo@npm:0.2.2`,
		`dts-bundle-generator@https://github.com/bluelovers/dts-bundle-generator.git#commit=2d724ed8a70fea6875699553ebc9173546e6a2e7`,
		`@strictsoftware/typedoc-plugin-monorepo@patch:@strictsoftware/typedoc-plugin-monorepo@npm%3A0.2.2#./.patches/@strictsoftware/typedoc-plugin-monorepo.patch::version=0.2.2&hash=916087&locator=%40yarnpkg%2Fgatsby%40workspace%3Apackages%2Fgatsby`,
	].forEach(rawInput => {
		test(rawInput, () =>
		{
			let actual = parseResolutionOrDescriptor(rawInput);
			expect(actual).toMatchSnapshot();

		});
	})

})
