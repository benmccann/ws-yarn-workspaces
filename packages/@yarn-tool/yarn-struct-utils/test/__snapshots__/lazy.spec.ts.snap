// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`lazy.spec @strictsoftware/typedoc-plugin-monorepo@npm:0.2.2 1`] = `
Object {
  "description": "npm:0.2.2",
  "fullName": "@strictsoftware/typedoc-plugin-monorepo",
  "isResolution": true,
  "parsed": Object {
    "descriptor": Object {
      "description": "npm:0.2.2",
      "fullName": "@strictsoftware/typedoc-plugin-monorepo",
    },
  },
  "rawInput": "@strictsoftware/typedoc-plugin-monorepo@npm:0.2.2",
}
`;

exports[`lazy.spec @strictsoftware/typedoc-plugin-monorepo@patch:@strictsoftware/typedoc-plugin-monorepo@npm%3A0.2.2#./.patches/@strictsoftware/typedoc-plugin-monorepo.patch::version=0.2.2&hash=916087&locator=%40yarnpkg%2Fgatsby%40workspace%3Apackages%2Fgatsby 1`] = `
Object {
  "description": "0.2.2",
  "fullName": "@strictsoftware/typedoc-plugin-monorepo",
  "isResolution": false,
  "parsed": Object {
    "descriptorHash": "bc7c13cee16e2ad09f7f741260c2374e932f33ced5aa58c0f46de97da51f5cb9897a79f3b0ab973de6f8178fbdf49a56f92c57e58e22a23995721642a765ed8e",
    "identHash": "f8d22e3172607f093b9122ee28ec17515bc2c0361abf41c54d4a1c474f685f2650d94af1b9fc8be7921345b5f2efe299255863a3f9cc0894604ea16ea305faab",
    "name": "typedoc-plugin-monorepo",
    "range": "patch:@strictsoftware/typedoc-plugin-monorepo@npm%3A0.2.2#./.patches/@strictsoftware/typedoc-plugin-monorepo.patch::version=0.2.2&hash=916087&locator=%40yarnpkg%2Fgatsby%40workspace%3Apackages%2Fgatsby",
    "scope": "strictsoftware",
  },
  "parsedRange": Object {
    "params": Object {
      "hash": "916087",
      "locator": "@yarnpkg/gatsby@workspace:packages/gatsby",
      "version": "0.2.2",
    },
    "protocol": "patch:",
    "selector": "./.patches/@strictsoftware/typedoc-plugin-monorepo.patch",
    "source": "@strictsoftware/typedoc-plugin-monorepo@npm:0.2.2",
  },
  "rawInput": "@strictsoftware/typedoc-plugin-monorepo@patch:@strictsoftware/typedoc-plugin-monorepo@npm%3A0.2.2#./.patches/@strictsoftware/typedoc-plugin-monorepo.patch::version=0.2.2&hash=916087&locator=%40yarnpkg%2Fgatsby%40workspace%3Apackages%2Fgatsby",
}
`;

exports[`lazy.spec dts-bundle-generator@https://github.com/bluelovers/dts-bundle-generator.git#commit=2d724ed8a70fea6875699553ebc9173546e6a2e7 1`] = `
Object {
  "description": undefined,
  "fullName": "dts-bundle-generator",
  "isResolution": false,
  "parsed": Object {
    "descriptorHash": "89f5f110d33f5edca7c707c84990988800eb4e8c474c3e67f9b2d81c99ad60622491d7997e3f88be3f5adbf7da11fb79a72917ffc03e7e148cd8d11bdb83d349",
    "identHash": "80f24fb2cb708bf649ca12e6d44b063ddd6d2add41ba056b7d355a0fe40dee672f63ece1a65c2b256efe8a3e77da59c61e8c43c59e5fb3406d66c64c73808e6c",
    "name": "dts-bundle-generator",
    "range": "https://github.com/bluelovers/dts-bundle-generator.git#commit=2d724ed8a70fea6875699553ebc9173546e6a2e7",
    "scope": null,
  },
  "parsedRange": Object {
    "params": null,
    "protocol": "https:",
    "selector": "commit=2d724ed8a70fea6875699553ebc9173546e6a2e7",
    "source": "//github.com/bluelovers/dts-bundle-generator.git",
  },
  "rawInput": "dts-bundle-generator@https://github.com/bluelovers/dts-bundle-generator.git#commit=2d724ed8a70fea6875699553ebc9173546e6a2e7",
}
`;
